zerver/actions/scheduled_messages.py

```diff
@@ -263,24 +305,36 @@ def try_deliver_one_scheduled_message(logger: logging.Logger) -> bool:
 
     try:
         send_scheduled_message(scheduled_message)
-    except JsonableError as e:
-        scheduled_message.failed = True
-        scheduled_message.failure_message = e.msg
-        scheduled_message.save(update_fields=["failed", "failure_message"])
-        logger.info("Failed with message: %s", scheduled_message.failure_message)
-    except Exception:
-        # An unexpected failure; store as a generic 500 error.
+    except Exception as e:
         scheduled_message.refresh_from_db()
         was_delivered = scheduled_message.delivered
         scheduled_message.failed = True
-        scheduled_message.failure_message = _("Internal server error")
+
+        if isinstance(e, JsonableError):
+            scheduled_message.failure_message = e.msg
+            logger.info("Failed with message: %s", e.msg)
+        else:
+            # An unexpected failure; store and send user a generic
+            # internal server error in notification message.
+            scheduled_message.failure_message = _("Internal server error")
+            logger.exception(
+                "Unexpected error sending scheduled message %s (sent: %s)",
+                scheduled_message.id,
+                was_delivered,
+                stack_info=True,
+            )
```

Comment:
This needs a `with override_language(user_profile.default_language)` around it, since it's run in a worker, doesn't have a current user, and we want the `_("Internal server error")` as well as the JsonableError possibly raised by `send_scheduled_message` to be in the user's language.

This has the side effect of making the error in our logs also be translated, but I don't think there's a simple way around that, because the JsonableError doesn't store the untranslated string.  We could potentially add that, which would let us defer the translation of this string to display time.

Reply:
Yeah, that sounds correct to me. If we care for our logs, we can add the exception `code` to what we store in the database in the future. And to be clear, the block that needs the override is the full `try` block.

Reply:
Yeah, my original manual testing for translated strings was flawed. Thanks for pointing this out.

I've updated these changes to that the full try/except block has the override for the sending user's language. Have screenshot of the Notification Bot direct message and log with an existing translated string in the easily triggered "this message send time is too far past" JsonableError in [this comment](https://github.com/zulip/zulip/pull/25553#issuecomment-1546146873).

Should I put the adding the exception `code` into the scheduled message model as a follow up?